### DISCUSSION TITLE
Is it possible to append to an array ? ( without  overriding the whole component config)

### DISCUSSION THREAD

**Discussion by joeitu** [LinkedSoftwareDependencies/Components.js#112] [Q&A] [answered]
Hello, 

 In CSS, `urn:solid-server:default:StaticAssetHandler` in [`config/http/static/default.json`](    https://github.com/CommunitySolidServer/CommunitySolidServer/blob/ba19a42a194cb3d70e0674a404e43c07e9cc24ef/config/http/static/default.json#L10) defines an array of "assets" that act as a key to value mapper.
 
 I would like to add a new key/value to this array. My current solution is to Override the whole `urn:solid-server:default:StaticAssetHandler` with my new key/value added to the array.
 However, I was wondering if it was possible to simply add the key/value to the array without overriding the whole component configuration?

**Comment by rubensworks** [comment, marked as answer]
No, that's not possible at the moment. Using the override system is the only option for this currently.

**Reply by joeitu** [reply]
Could this be a legit feature request? 

The motivation is when you have to write two separate components that both need to add a key/value to the array. 

For example with two CSS components:
`ModuleOne` need to add a js script to the `well_known` folder. To achieve that it will override  `urn:solid-server:default:StaticAssetHandler`
`ModuleTwo` needs to do the same.
When adding both `ModuleOne` and `ModuleTwo` to CSS graph, only the script from `ModuleTwo` will be available in the `well_known` folder. Because `ModuleOne` override will be in turn overridden by `MoculeTwo`

Hope that makes sense :)

**Reply by rubensworks** [reply]
Sure, definitely makes sense. Feel free to make an issue. @joachimvh might also be interested in this.

**Reply by joachimvh** [reply]
You can append in Components.js though, it's just not "officially" supported and the resulting order is undefined. We use it in several places in the CSS configs. E.g. here we add the registration HTML page to the list of pages if registration is enabled: https://github.com/CommunitySolidServer/CommunitySolidServer/blob/ba19a42a194cb3d70e0674a404e43c07e9cc24ef/config/identity/registration/enabled.json#L24-L33

**Reply by joeitu** [reply]
You mean the list of pages defined here : https://github.com/CommunitySolidServer/CommunitySolidServer/blob/ba19a42a194cb3d70e0674a404e43c07e9cc24ef/config/identity/handler/interaction/views/html.json#L26 ?

**Reply by joachimvh** [reply]
Yes, the `templates` parameter of `urn:solid-server:auth:password:HtmlViewHandler` will receive a combination of those configurations as input.

**Reply by joeitu** [reply]
Awesome, I'll take inspiration from that
